SpringBoot demo

deb05121 opened the pull request.

deb05121 committed:
SpringBoot demo

deb05121 committed:
refactored demo project

deb05121 committed:
tabula rasa

deb05121 committed:
update

lyancsie commented on demo/src/main/java/com/example/demo/controller/WebshopRestController.java:
```diff
@@ -0,0 +1,40 @@
+
+    @Autowired
+    //private WebshopService webshopService;
+    private WebshopRepository webshopRepository;
```
Controllers should not contain dependencies from the repo layer

deb05121 replied:
I changed it to the next line:
private final WebshopService webshopService;

lyancsie commented on demo/src/main/java/com/example/demo/controller/WebshopRestController.java:
```diff
@@ -0,0 +1,40 @@
+    //private WebshopService webshopService;
+    private WebshopRepository webshopRepository;
+
+    @GetMapping("/getAllItem")
```
```suggestion
    @GetMapping("/getAllItems")
```

deb05121 replied:
I fixed it.

lyancsie commented on demo/src/main/java/com/example/demo/controller/WebshopRestController.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+
+    @GetMapping("/getItemByName")
```
```suggestion
    @GetMapping("/getItemsByName")
```

deb05121 replied:
It's ready.

lyancsie commented on demo/src/main/java/com/example/demo/controller/WebshopRestController.java:
```diff
@@ -0,0 +1,40 @@
+        return webshopRepository.findByNameOrName(name1, name2);
+    }
+
+    @GetMapping("/getItemByPrice")
```
```suggestion
    @GetMapping("/getItemsByPrice")
```

deb05121 replied:
It's also done.

lyancsie commented on demo/src/main/java/com/example/demo/controller/WebshopRestController.java:
```diff
@@ -0,0 +1,40 @@
+        return webshopRepository.findByPrice(price);
+    }
+
+    @GetMapping("/getItemByPriceAndQuantity")
```
```suggestion
    @GetMapping("/getItemsByPriceAndQuantity")
```

deb05121 replied:
It is fixed, too.

lyancsie commented on demo/src/main/java/com/example/demo/service/WebshopService.java:
```diff
@@ -0,0 +1,13 @@
+import java.util.List;
+
+public class WebshopService {
+
```
you should autowire the and do all the business logic here

deb05121 replied:
I guess, I managed to fix it.

lyancsie replied:
good

lyancsie commented on demo/src/main/resources/templates/webshop.html:
```diff
@@ -0,0 +1,10 @@
+<body>
+
+</body>
+</html>
```
Why do we need this?

lyancsie replied:
still waiting for the answer

deb05121 replied:
I deleted <body> tag.

lyancsie replied:
@deb05121 no answer received

deb05121 replied:
@lyancsie here the body tag is unnecessary, so I deleted that.

lyancsie replied:
@deb05121 The whole html page is unnecessary, since you are having `REST` end points and you are returning JSONs, not HTML pages.

lyancsie replied:
HTML pages are needed when you are using controllers, not rest controllers and you want to return dynamically created web pages

lyancsie replied:
@deb05121 please address this - you don't need this html at all

deb05121 replied:
I deleted html page from resources, because I use just rest controller with JSONs, as returning.

lyancsie commented on demo/mvnw.cmd:
```diff
@@ -0,0 +1,205 @@
+
+if "%MAVEN_TERMINATE_CMD%"=="on" exit %ERROR_CODE%
+
+cmd /C exit /B %ERROR_CODE%
```
just out of curiosity - how did these things occur here?

lyancsie replied:
I know what these things are :) But please make sure to do your research and answer this question before merging these changes in

lyancsie replied:
@deb05121 no answer received

deb05121 replied:
@lyancsie It shows how to stop Maven execution
when it stops, it automatically generates an error code

lyancsie replied:
i didn't mean that exact line, but the whole 200-line file :) anyways, if you are using maven wrapper, be sure to understand what that is

deb05121 replied:
The maven wrapper installs a more compact version of maven for a specific project.
"mvnw files are not necessarily part of our projects. However, including them could be beneficial."

lyancsie reviewed (changes requested):
Multiple violations that need to be dealt with. Next time, please link your pr in the ticket... The branch name is also odd, you are not refactoring anything

deb05121 committed:
I refactored the layers and the tests

deb05121 committed:
I fixed endpoints' names

lyancsie commented on demo/src/main/java/com/example/demo/controller/WebshopRestController.java:
```diff
@@ -0,0 +1,46 @@
+
+    @GetMapping("/getItemsByPrice")
+    List<ShopItem> getItemsByPrice(@RequestParam int price) {
+        log.info("Finding item by price {}", price);
```
Please fix the logs (here and everywhere)

deb05121 replied:
"Finding items by price {}"
because it may be more than one item as result

lyancsie commented on demo/src/test/java/com/example/demo/controller/ShopItemTest.java:
```diff
@@ -0,0 +1,106 @@
+                        .quantity(quantityOfItem)
+                        .date(null)
+                        .build()));
+        final var shopItems = webshopRepository.findByPriceAndQuantity(priceOfItem, quantityOfItem);
```
This does not make any sense. You are mocking the repo and after asserting for the same thing - please fix your tests. Your test name is also odd and you are not testing the controller, so don't put it in the controller layer :)

deb05121 replied:
I replaced and rename this test.

lyancsie replied:
You have replaced and renamed your test - that's good; but you are still not testing your service! Remember: you want to test the service individually. Then you have to mock everything that the service depends on (in this case the repo) and test the service in an isolated way.

Now you are mocking the repo and subsequently testing the repo - which doesn't make any sense.

lyancsie commented on demo/src/main/resources/application.properties:
```diff
@@ -0,0 +1,6 @@
+spring.application.name=springboot-demo
+spring.datasource.url=jdbc:mysql://localhost:3306/webshop
+spring.datasource.username=root
+spring.datasource.password=dEb05121
```
You have included application.properties with sensitive information - this is considered to be a very serious violation. Please remove this file

deb05121 replied:
@lyancsie 
I wrote "application.properties" into the gitignor, but I had to delete by own hand this file from demo project

lyancsie replied:
application.properties is still there. please remove it

lyancsie commented on demo/src/main/java/com/example/demo/service/WebshopService.java:
```diff
@@ -0,0 +1,48 @@
+    public List<ShopItem> findByPriceAndQuantity(int price, int quantity){
+        final var items = webshopRepository.findByPriceAndQuantity(price, quantity);
+        if(items.isEmpty()){
+            throw new ItemNotFoundException("No item is found with price " + price + " and quantity " + quantity + "\n");
```
These \n characters here and in the other exceptions are unnecessary - please get rid of them

deb05121 replied:
@lyancsie 
I deleted \n characters from all exceptions.

lyancsie replied:
good

deb05121 committed:
screen shots

I apologize for the delay.
There are some photos of server running and Postman results.

lyancsie commented:
@deb05121  good - but please don't put the screen shots in the code - please paste these types of screen shots in the Jira ticket and delete the images from here

deb05121 committed:
Delete WebshopServerRunning.png

deb05121 committed:
Delete getAllItems_200_OK.png

deb05121 committed:
Delete getItemsByPriceAndQuantity_200_OK.png

deb05121 committed:
Delete getItemsByPriceAndQuantity_500_Error.png

deb05121 committed:
Delete getItemsByPrice_200_OK.png

deb05121 committed:
Delete getItemsByPrice_500_Error.png

deb05121 committed:
Delete getItemsByTwoNames_200_OK.png

deb05121 committed:
Delete getItemsByTwoNames_500_Error.png

deb05121 committed:
I fixed all my fault

deb05121 committed:
Merge branch 'feature-COMPARATOR-280-refactor' of https://github.com/deb05121/SpringBootProjects into feature-COMPARATOR-280-refactor

lyancsie commented on demo/src/test/java/com/example/demo/service/ShopItemServiceTest.java:
```diff
@@ -0,0 +1,105 @@
+        Assertions.assertEquals(nameOfItemA, shopItems.get(0).getName());
+        Assertions.assertEquals(nameOfItemB, shopItems.get(1).getName());
+    }
+
```
your tests still do not test what they should - please review carefully and adjust your tests

deb05121 replied:
It is partially solved, it will be continued.

lyancsie commented on demo/src/main/java/com/example/demo/service/WebshopService.java:
```diff
@@ -0,0 +1,48 @@
+    public List<ShopItem> findByTwoNames(String name1, String name2){
+        final var items = webshopRepository.findByNameOrName(name1, name2);
+        if(items.isEmpty()){
+            throw new ItemNotFoundException("No item is found with name " + name1 + " and name " + name2);
```
`and` or `or`? :)

deb05121 replied:
@lyancsie 
I rewrote this line in code:
throw new ItemNotFoundException("No item is found with name " + name1 + " or name " + name2);

lyancsie replied:
@deb05121 odd, because I still do not see the change
![image](https://github.com/deb05121/SpringBootProjects/assets/42215450/82ac6a43-7085-4c6a-9337-eb65b6ad0ca7)

lyancsie commented on demo/src/main/java/com/example/demo/service/WebshopService.java:
```diff
@@ -0,0 +1,48 @@
+
+    public List<ShopItem> findAllItems(){
+        final var items = webshopRepository.findAll();
+        if(items.isEmpty()){
```
autoformat issues..

deb05121 replied:
@lyancsie 
Ctrl + Alt + L
This is done.

lyancsie replied:
it might be done, but it is not pushed yet :)

deb05121 replied:
It was pushed yesterday night.

deb05121 committed:
update, delete, put were added

deb05121 committed:
Update .gitignore

application.properties was added

deb05121 committed:
repeated

deb05121 committed:
Merge branch 'main' into feature-COMPARATOR-280-refactor

deb05121 committed:
Update .gitignore

lyancsie commented on demo/src/test/java/com/example/demo/controller/ShopItemTest.java:
```diff
@@ -0,0 +1,106 @@
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class ShopItemTest {
```
```suggestion
class WebhopServiceTest {
```

deb05121 replied:
I renamed this class. I will push soon.

deb05121 committed:
mod service and service test, del html

deb05121 committed:
test was renamed WebshopServiceTest
